### PR TITLE
Course enrollment

### PR DESCRIPTION
**Pull request by xenoteo**

### CODE REVIEW

**Review by marcinz99 — APPROVED** [review]
Nice. Seems to work fine. Didn't know about thymeleaf fragments btw.

File: src/main/java/upeldev/com/github/upel3/services/CourseService.java [hunk 1]
```diff
@@ -68,5 +70,23 @@ public void addLecturer(Course course, User lecturer){
...
+
+    public void addStudent(User student, String courseCode){
+        Course course = findCourseByAccessCode(courseCode);
+        if (course == null){
+            throw new IllegalArgumentException("No such course");
+        }
+        if (course.getEnrolledStudents().contains(student)){
+            throw new IllegalArgumentException("Student it already enrolled to this course");
+        }
+        addStudent(course, student);
```
Comment (anosek-an) [inline src/main/java/upeldev/com/github/upel3/services/CourseService.java]: Rename one of the methods or make addStudent that doesn't throw exceptions private.

File: src/main/java/upeldev/com/github/upel3/controllers/WebController.java [hunk 2]
```diff
@@ -96,4 +96,24 @@ public String profile(Model model, Principal principal) {
...
+        User currentUser = userService.findByEmail(principal.getName());
+        model.addAttribute("user", currentUser);
+        return "course_enrollment";
+    }
+
+    @RequestMapping(value = "/enroll")
+    public String enroll(
+            @RequestParam(value = "courseCode") String courseCode, Model model, Principal principal) {
+        User currentUser = userService.findByEmail(principal.getName());
+        courseService.addStudent(currentUser, courseCode);
```
Comment (anosek-an) [inline src/main/java/upeldev/com/github/upel3/controllers/WebController.java]: addStudent throws exceptions that aren't caught here

**Review by anosek-an — CHANGES_REQUESTED** [review]
Add exception handling.
Thanks for the quick implementation!

**Review by anosek-an — APPROVED** [review]
Thanks for the changes. Great job!

**Pull request merged**